internal/releasesources/release_sources.go

```diff
@@ -503,3 +506,58 @@ func getReleaseImageReference(version, cpuArchitecture string) string {
 	}
 	return fmt.Sprintf(releaseImageReferenceTemplateSaaS, version, cpuArchitecture)
 }
+
+func RunOpenshiftReleaseSyncerIfNeeded(
+	releaseSources models.ReleaseSources,
+	releaseImages models.ReleaseImages,
+	db *gorm.DB,
+	log logrus.FieldLogger,
+	releaseHandler oc.Release,
+	leader,
+	startupLeader leader.ElectorInterface,
+	enableKubeAPI bool,
+	releaseSourcesConfig Config,
+	failOnError func(err error, msg string, args ...interface{}),
+) *thread.Thread {
+	if !enableKubeAPI && releaseSourcesConfig.ReleaseSources != "" {
+		releaseSourcesHandler, err := newReleaseSourcesHandler(
+			releaseSources, releaseImages, log, db, releaseSourcesConfig, leader,
+		)
+		if err != nil {
+			failOnError(err, "error occurred while creating release sources handler")
+		}
+
+		failOnError(startupLeader.RunWithLeader(context.Background(), func() error {
```

Comment:
Is failOnError function panicking? shouldn't this return an error and let the main fail?